hypothesis-python/src/hypothesis/stateful.py

```diff
@@ -77,7 +77,7 @@ def is_breaking_run(runner):
             return False
         except HypothesisException:
             raise
-        except Exception:
+        except failure_exceptions_to_catch:
```

Comment:
This looks wrong: you’re catching the function. Don’t you mean to match on `exceptions_to_catch`?